Source/WebKit/UIProcess/WebProcessProxy.cpp

```diff
@@ -2556,38 +2557,48 @@ void WebProcessProxy::getNotifications(const URL& registrationURL, const String&
     WebNotificationManagerProxy::sharedServiceWorkerManager().getNotifications(registrationURL, tag, sessionID(), WTFMove(callback));
 }
 
-std::optional<Vector<uint8_t>> WebProcessProxy::getWebCryptoMasterKey()
+void WebProcessProxy::getWebCryptoMasterKey(CompletionHandler<void(std::optional<Vector<uint8_t>>&&)>&& completionHandler)
 {
-    if (auto isKey = m_websiteDataStore->client().webCryptoMasterKey())
-        return isKey;
-    if (auto isKey = defaultWebCryptoMasterKey())
-        return isKey;
-    return std::nullopt;
+    if (auto isKey = m_websiteDataStore->client().webCryptoMasterKey()) {
+        completionHandler(WTFMove(isKey));
+        return;
+    }
+    if (!m_websiteDataStore->client().isAsyncMasterKeyDelegateSet()) {
+        if (auto isKey = defaultWebCryptoMasterKey())
+            completionHandler(WTFMove(isKey));
+        else
+            completionHandler(std::nullopt);
+        return;
+    }
```

Comment:
If we fall back to default key in `webCryptoMasterKeyAsync` when the new delegate method is not specified, then we don't need this check and we could simplify the change in NavigationState.mm. 
We might do it in a followup.

Reply:
sure.. lets get the Safari side done first.. this could be done later!